Feature/088

JaimeBorregoConde opened the pull request:
Diseño de censo, concretamente el diseño para la pagina de exportación censo en archivo de tipo excel

gonmarfer2 committed:
feat: added CensusGroups

Remade the model to add groups of censuses to filter with.

gonmarfer2 committed:
feat: intermediate commit for census views

Tried to make some views, permissions not working

gonmarfer2 committed:
feat: census groups

Modified views, model and other relevant files.

gonmarfer2 committed:
test,fix: census groups

Also fixed the views for Census and CensusGroup to add missing fragments.

JaimeBorregoConde committed:
FEAT: added the import of excel files to census

Added in two different ways. A manual one, where a user as a staff can import an excel, and an automatic one managed by a library to import files in administrator panel

gonmarfer2 committed:
fix: groups api works fine now

I had to redo everything because I didn't properly understand how everything works

gonmarfer2 committed:
fix: changed permission class to IsAdminUser

Because the one given doesn't work (request.auth is always None). This one seems equivalent.

gonmarfer2 committed:
fix: census can now be pushed without a group

Bug found on the post method.

gonmarfer2 committed:
test: redone some tests to make them work again

Changed JSON data structure to avoid failures and changed parameters.

AntoCampos12 committed:
FEAT: FEATURE/026

· Now we can group census
· Some bugs with user, admin permissions was fixed
· Census was modified, fixing some methods and including serializers

JaimeBorregoConde committed:
FIX: change import_excel view

Import_excel has been fixed due to changes in Cesus model. Furthermmore, url for importing users excel has been modified from cesus/import_excel/ to /cesus/import.

JaimeBorregoConde committed:
FEAT: adding export census

Export census has been added for Staff users. Yo may enter /census/export to use it

JaimeBorregoConde committed:
test: test for testing export file
Functional test to make sure excel should be what census have.

JaimeBorregoConde committed:
Fix: modified test and django.yml for actions
 Test modified to be a little bit more complex. Added makemigrations in django.yml for actions in github

JaimeBorregoConde committed:
Fix: Modified export excel method
 Added message test and hcenged url and class due to the fact that it was inheriting from API

JaimeBorregoConde committed:
Merge branch 'feature/029' of https://github.com/egc-decide-trebujena/decide into feature/112

JaimeBorregoConde committed:
Feat: Change export.html template for translation use and for having a first export view
 The view has been change using import excel template as a model.

gonmarfer2 reviewed (changes requested):
No has puesto la etiqueta de traducción en *excel.html*

JaimeBorregoConde committed:
Fix: Added translation labels due to reviewer
 I had misunderstood how to put translation labels in desing issues. Fixed. Fuerthermore, excel.html was a residual file. It has been deleted.
